metrics/core/test/util/validation/printer/validation_result_printer_test.dart

```diff
@@ -0,0 +1,276 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:metrics_core/src/util/validation/printer/validation_result_printer.dart';
+import 'package:metrics_core/src/util/validation/target_validation_result.dart';
+import 'package:metrics_core/src/util/validation/validation_conclusion.dart';
+import 'package:metrics_core/src/util/validation/validation_result.dart';
+import 'package:metrics_core/src/util/validation/validation_target.dart';
+
+import 'package:test/test.dart';
+
+// ignore_for_file: avoid_redundant_argument_values
+
+void main() {
+  group("ValidationResultPrinter", () {
+    const target = ValidationTarget(name: 'service 1', description: 'test');
+    const secondTarget = ValidationTarget(
+      name: 'service 2',
+      description: 'test',
+    );
+    const conclusion = ValidationConclusion(name: 'success', indicator: '+');
+    const details = {'test': 'details'};
+    const context = {'test': 'context'};
+    const firstResult = TargetValidationResult(
+      target: target,
+      conclusion: conclusion,
+      description: 'description 1',
+      details: details,
+      context: context,
+    );
+    const secondResult = TargetValidationResult(
+      target: secondTarget,
+      conclusion: conclusion,
+      description: 'description 2',
+      details: details,
+      context: context,
+    );
+
+    StringSink sink;
+    ValidationResultPrinter resultPrinter;
+
+    Matcher containsAllEntries(Map<String, dynamic> fromMap) {
+      final entries = <String>[];
+      fromMap.forEach((key, value) {
+        entries.add(key);
+        entries.add(value.toString());
+      });
+
+      return stringContainsInOrder(entries);
+    }
+
+    Matcher validationResultMessageMatcher(
```

Comment:
🔖 
```suggestion
    Matcher matchesValidationResultMessage(
```
Or maybe we can use else name - the main idea is to make it closer to existing matchers that are likely a verb or contains a verb.